packages/data-stores/src/queries/use-jetpack-search-ai.ts

```diff
@@ -0,0 +1,45 @@
+import { useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+
+type AIResponseURL = {
+	url: string;
+	title: string;
+};
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
+export type JetpackSearchAIResult = {
+	response: string;
+	step: string;
+	urls: AIResponseURL[];
+	terms: string[];
+};
+
+export function useJetpackSearchAIQuery( siteId: number | string, query: string, stopAt: string ) {
+	return useQuery< JetpackSearchAIResult >(
+		[ 'aiQuery', query, stopAt ],
+		async () =>
+			canAccessWpcomApis()
+				? await wpcomRequest( {
```

Comment:
This creates a promise that wraps a promise. 

```suggestion
	 () =>
			canAccessWpcomApis()
				? wpcomRequest( {
```